watertap/property_models/activated_sludge/asm1_reactions.py

```diff
@@ -0,0 +1,550 @@
+            domain=pyo.PositiveReals,
+            doc="Oxygen half-saturation coefficient for heterotrophic biomass, K_O,H",
+        )
+        self.half_saturation_coeff_oxygen_autotrophic = pyo.Var(
+            initialize=0.4e-3,
+            units=pyo.units.kg / pyo.units.m**3,
+            domain=pyo.PositiveReals,
+            doc="Oxygen half-saturation coefficient for autotrophic biomass, K_O,A",
+        )
+        self.half_saturation_coeff_nitrate_heterotrophic = pyo.Var(
+            initialize=0.5e-3,
+            units=pyo.units.kg / pyo.units.m**3,
+            domain=pyo.PositiveReals,
+            doc="Nitrate half-saturation coefficient for denitrifying heterotrophic biomass, K_NO",
+        )
+        self.decay_coeff_heterotrophic = pyo.Var(
+            initialize=0.3,
+            units=1 / pyo.units.day,
+            domain=pyo.PositiveReals,
+            doc="Decay coefficient for heterotrophic biomass, b_H",
+        )
+        self.decay_coeff_autotrophic = pyo.Var(
+            initialize=0.05,
+            units=1 / pyo.units.day,
+            domain=pyo.PositiveReals,
+            doc="Decay coefficient for autotrophic biomass, b_A",
+        )
+        self.eta_g = pyo.Var(
+            initialize=0.8,
+            units=pyo.units.dimensionless,
+            domain=pyo.PositiveReals,
+            doc="Correction factor for mu_H under anoxic conditions, eta_g",
+        )
+        self.eta_h = pyo.Var(
+            initialize=0.8,
+            units=pyo.units.dimensionless,
+            domain=pyo.PositiveReals,
+            doc="Correction factor for hydolysis under anoxic conditions, eta_h",
+        )
+        self.max_hydrolysis_rate = pyo.Var(
+            initialize=3,
+            units=1 / pyo.units.day,
+            domain=pyo.PositiveReals,
+            doc="Maximum specific hydrolysis rate, k_h",
+        )
+        self.half_saturation_coeff_hydrolysis = pyo.Var(
+            initialize=0.1,
+            units=pyo.units.dimensionless,
+            domain=pyo.PositiveReals,
+            doc="Half-saturation coefficient for hydolysis of slowly biodegradable substrate, K_X",
+        )
+        self.half_saturation_coeff_ammonium = pyo.Var(
+            initialize=1e-3,
+            units=pyo.units.kg / pyo.units.m**3,
+            domain=pyo.PositiveReals,
+            doc="Ammonia Half-saturation coefficient for autotrophic biomass, K_NH",
+        )
+        self.ammonification_rate = pyo.Var(
+            initialize=0.05e3,
+            units=pyo.units.m**3 / pyo.units.kg / pyo.units.day,
+            domain=pyo.PositiveReals,
+            doc="Ammonification rate, k_a",
+        )
+
+        # Reaction Stoichiometry
+        # This is the stoichiometric part the Peterson matrix in dict form
+        # Note that the coeffiicents for alkalinity are givein on a mass basis
+        # Conversion to molar basis is handled elsewhere for unit consistency
+        self.rate_reaction_stoichiometry = {
```

Comment:
Something that isn't immediately clear to me is where this is being utilized

Reply:
It gets used in the control volumes to convert extent of each reaction into amount of species generated by each reaction.